<launch>
  <arg name="robot_name" default="robot_m"/>

  <remap from="tag_detections" to="/$(arg robot_name)/tag_detections"/>
  <remap from="tag_detections_image" to="/$(arg robot_name)/tag_detections_image"/>
  <remap from="tag_detections_pose" to="/$(arg robot_name)/tag_detections_pose"/>

  <node pkg="apriltags_ros" type="apriltag_detector_node" name="$(arg robot_name)_apriltag_detector" output="screen">
    <!-- Remap topic required by the node to custom topics -->
    <remap from="image_rect" to="/$(arg robot_name)/camera/rgb/image_raw" />
    <remap from="camera_info" to="/$(arg robot_name)/camera/rgb/camera_info" />

    <!-- Optional: Subscribe to the compressed stream -->
    <param name="image_transport" type="str" value="compressed" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <!-- Enable projected optical measurements for more accurate tag transformations -->
    <!-- This exists for backwards compatability and should be left true for new setups -->
    <param name="projected_optics" type="bool" value="true" />

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions" subst_value="True">[
      {id: 0, size: 0.163513, frame_id: $(arg robot_name)/tag_000},
      {id: 1, size: 0.163513, frame_id: $(arg robot_name)/tag_001},
      {id: 2, size: 0.163513, frame_id: $(arg robot_name)/tag_002},
      {id: 3, size: 0.163513, frame_id: $(arg robot_name)/tag_003},
      {id: 4, size: 0.163513, frame_id: $(arg robot_name)/tag_004},
      {id: 5, size: 0.163513, frame_id: $(arg robot_name)/tag_005},
      {id: 6, size: 0.163513, frame_id: $(arg robot_name)/tag_006},
      {id: 7, size: 0.163513, frame_id: $(arg robot_name)/tag_007},
      ]
    </rosparam>
  </node>
</launch>